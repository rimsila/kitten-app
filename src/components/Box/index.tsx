import { createBox } from '@shopify/restyle';
import { AppTheme } from 'themes';

const Box = createBox<AppTheme>();
export default Box;
