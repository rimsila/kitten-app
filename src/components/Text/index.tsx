import { createText } from '@shopify/restyle';
import { AppTheme } from 'themes';

const Text = createText<AppTheme>();
export default Text;
