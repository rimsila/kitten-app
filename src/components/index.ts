export { default as CusBox } from './Box';
export { default as CusButton } from './Button';
export { default as Card } from './Card';
export { default as Spinning } from './FeedBack/Spining';
export { default as CusText } from './Text';
