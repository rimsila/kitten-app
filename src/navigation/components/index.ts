export { default as FocusAwareStatusBar } from './FocusAwareStatusBar';
