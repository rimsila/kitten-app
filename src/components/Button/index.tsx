import { CusBox, CusText } from 'components';
import React, { ReactNode } from 'react';
import { TouchableOpacity } from 'react-native';

export type ButtonProps = React.ComponentProps<typeof CusBox> & {
  onPress?: () => void;
  label?: string;
  children?: ReactNode;
  labelProps?: React.ComponentProps<typeof CusText>;
};

const Button = ({ onPress, label, children, labelProps, ...rest }: ButtonProps) => {
  return (
    <TouchableOpacity onPress={onPress}>
      <CusBox {...rest}>
        {typeof label === 'string' ? (
          <CusText variant="subheader" {...labelProps}>
            {label}
          </CusText>
        ) : (
          children
        )}
      </CusBox>
    </TouchableOpacity>
  );
};

export default Button;
