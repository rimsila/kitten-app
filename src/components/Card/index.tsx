import { createRestyleComponent, createVariant, VariantProps } from '@shopify/restyle';
import { CusBox } from 'components';
import { AppTheme } from 'themes';

const Card = createRestyleComponent<
  VariantProps<AppTheme, 'cardVariants'> & React.ComponentProps<typeof CusBox>,
  AppTheme
>([createVariant({ themeKey: 'cardVariants' })], CusBox);

export default Card;
