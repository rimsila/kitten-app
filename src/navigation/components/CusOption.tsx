import { CusBox, CusText } from 'components';
import Button from 'components/Button';
import { useNetWork } from 'hooks/useNetwork';
import { goBack, isCanGoBack } from 'navigation/NavigationService';
import React, { ReactNode } from 'react';
import { ifIphoneX } from 'utils/dimension';

export type CusHeaderProps = {
  center?: string | boolean | ReactNode;
  left?: ReactNode | string | boolean;
  right?: ReactNode | boolean;
};

const CusHeader = (props: CusHeaderProps) => {
  const { netInfo } = useNetWork();
  const { center, left = true, right = true } = props;

  return (
    <CusBox
      bg="fgSuccess"
      height={50}
      alignItems="center"
      paddingHorizontal="l"
      flexDirection="row"
      mt={ifIphoneX('60', 'none')}
    >
      <CusBox flex={1} flexDirection="row" alignItems="center" justifyContent="flex-start">
        {typeof left === 'boolean' && left && isCanGoBack() ? (
          <Button onPress={goBack} flexDirection="row" alignItems="center">
            <CusText variant="subheader" color="fgContrasting" opacity={0.9}>
              &#60;&nbsp;
            </CusText>
            <CusText variant="body" color="fgContrasting" opacity={0.9}>
              Back
            </CusText>
          </Button>
        ) : (
          left
        )}
      </CusBox>

      <CusBox flex={1} justifyContent="center" alignItems="center">
        {typeof center === 'string' ? (
          <CusText variant="body" color="fgContrasting" opacity={0.9} textAlign="center">
            {center}
          </CusText>
        ) : (
          center
        )}
      </CusBox>

      <CusBox flex={1} justifyContent="flex-end">
        {typeof right === 'boolean' ? (
          <CusText variant="body" color="fgContrasting" opacity={0.9} textAlign="right">
            {netInfo?.isConnected ? 'Online' : 'Offline'}
          </CusText>
        ) : (
          right
        )}
      </CusBox>
    </CusBox>
  );
};

export default CusHeader;
