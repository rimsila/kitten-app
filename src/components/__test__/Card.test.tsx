import { render } from '@testing-library/react-native';
import { Card } from 'components';
import React from 'react';

describe('<Card />', () => {
  it('should match snapshot', () => {
    const rendered = render(<Card />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
