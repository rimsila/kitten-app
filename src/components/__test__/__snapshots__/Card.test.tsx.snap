// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "display": "flex",
      },
      Array [
        Object {},
      ],
    ]
  }
/>
`;
